pandas/core/arrays/arrow/array.py

```diff
@@ -1999,27 +2045,110 @@ def _groupby_op(
                 **kwargs,
             )
 
-        # maybe convert to a compatible dtype optimized for groupby
-        values: ExtensionArray
         pa_type = self._pa_array.type
-        if pa.types.is_timestamp(pa_type):
-            values = self._to_datetimearray()
-        elif pa.types.is_duration(pa_type):
-            values = self._to_timedeltaarray()
+        pa_supported_groupby_aggregations = [
+            "any",
+            "all",
+            "sum",
+            "prod",
+            "min",
+            "max",
+            "mean",
+            "sem",
+            "std",
+            "var",
+        ]
+        # aggregations not yet supported natively in pyarrow:
+        #  - first/last: https://github.com/apache/arrow/issues/36709
+        #  - nth
+        #  - median
+
+        if how not in pa_supported_groupby_aggregations:
+            # maybe convert to a compatible dtype for groupby
+            values: ExtensionArray
+            pa_type = self._pa_array.type
+            if pa.types.is_timestamp(pa_type):
+                values = self._to_datetimearray()
+            elif pa.types.is_duration(pa_type):
+                values = self._to_timedeltaarray()
+            else:
+                values = self._to_masked()
+
+            result = values._groupby_op(
+                how=how,
+                has_dropped_na=has_dropped_na,
+                min_count=min_count,
+                ngroups=ngroups,
+                ids=ids,
+                **kwargs,
+            )
+            if isinstance(result, np.ndarray):
+                return result
+            return type(self)._from_sequence(result, copy=False)
+
+        arr = self._get_values_for_reduction(how)
+
+        pa_name = {
+            "prod": "product",
+            "std": "stddev",
+            "var": "variance",
+        }.get(how, how)
+
+        # pyarrow does not accept a min_count of -1
+        if min_count == -1:
+            if how in ["any", "all"]:
+                min_count = 0
+            else:
+                min_count = 1
```

Comment:
Why the difference in min_count here?

Reply:
pandas defaults min_count to 0 whereas pyarrow defaults min_count to 1. This ensures we get the same result for grouped and non-grouped any/all. `ArrowExtensionArray._pyarrow_reduce` does the same. Without this the groupby resutl below would be `pd.NA` rather than `False`:

```
In [1]: import pandas as pd

In [2]: ser = pd.Series([pd.NA], dtype="boolean[pyarrow]")

In [3]: ser.any()
Out[3]: False

In [4]: ser.groupby(level=0).any()
Out[4]: 
0    False
dtype: bool[pyarrow]
```